adapter/internal/oasparser/model/async_api.go

```diff
@@ -0,0 +1,243 @@
+	swagger.securityScheme = asyncAPI.getSecuritySchemes()
+	swagger.resources = asyncAPI.getResources()
+
+	swagger.apiType = WS
+
+	if asyncAPI.Servers.Production.URL != "" {
+		endpoint, err := getEndpointForWebsocketURL(asyncAPI.Servers.Production.URL)
+		if err == nil {
+			productionEndpoints := append([]Endpoint{}, *endpoint)
+			swagger.productionEndpoints = generateEndpointCluster("clusterProd",
+				productionEndpoints, "load_balance")
+		} else {
+			return errors.New("error encountered when parsing the production endpoint for AsyncAPI")
+		}
+	}
+	if asyncAPI.Servers.Sandbox.URL != "" {
+		endpoint, err := getEndpointForWebsocketURL(asyncAPI.Servers.Sandbox.URL)
+		if err == nil {
+			sandboxEndpoints := append([]Endpoint{}, *endpoint)
+			swagger.sandboxEndpoints = generateEndpointCluster("clusterSand",
+				sandboxEndpoints, "load_balance")
+		} else {
+			return errors.New("error encountered when parsing the sandbox endpoint for AsyncAPI")
+		}
+	}
+	return nil
+}
+
+func (asyncAPI AsyncAPI) getSecuritySchemes() []SecurityScheme {
+	securitySchemes := []SecurityScheme{}
+	for key, securityScheme := range asyncAPI.Components.SecuritySchemes {
+		securityScheme.DefinitionName = key
+		securitySchemes = append(securitySchemes, securityScheme)
+	}
+	return securitySchemes
+}
+
+func (asyncAPI AsyncAPI) getResources() []*Resource {
+	resources := []*Resource{}
+	val, found := resolveAsyncAPIDisableSecurity(asyncAPI)
+	for channel, channelItem := range asyncAPI.Channels {
+		// ex: channel = /notify, channelItem = { Publish:map, Subscribe:map }
+		var pubOrSubVendorExtensions map[string]interface{}
+
+		if channelItem.Publish != nil {
+			pubOrSubVendorExtensions = channelItem.Publish.(map[string]interface{})
+			if channelItem.Subscribe != nil {
+				loggers.LoggerOasparser.Warnf(
+					"Both Publish and Subscribe types exists for the same topic. "+
+						"Prioritizing the extensions under type Publish for the topic %v.", channel)
+			}
+		} else if channelItem.Subscribe != nil {
+			pubOrSubVendorExtensions = channelItem.Subscribe.(map[string]interface{})
+		} else {
+			loggers.LoggerOasparser.Warnf(
+				"The topic does not include a Publish or a Subscribe definition. Discarding the topic %v.", channel)
+			continue
+		}
+
+		if found {
+			pubOrSubVendorExtensions["x-wso2-disable-security"] = val
+		}
+
+		if channelItem.XAuthType == "None" || channelItem.XWso2DisableSecurity {
+			pubOrSubVendorExtensions["x-wso2-disable-security"] = true
+		}
+
+		security := getSecurityArray(pubOrSubVendorExtensions)
+		operation := NewOperation("GET", security, pubOrSubVendorExtensions)
+
+		var methodsArray []*Operation
+		methodsArray = append(methodsArray, operation)
+
+		// we ignore other topic vendor extensions except x-auth-type and x-wso2-disable-security
+		channelVendorExtensions := map[string]interface{}{}
+		if found {
+			channelVendorExtensions["x-wso2-disable-security"] = val
+		}
```

Comment:
```suggestion
```
methodsArray already contains this value properly. We don't need to pass this value to resource-level extensions.